### PR TITLE
Git provider & add badges

### PR DESCRIPTION
**Pull request by DavidBadura**
Support for:

* gitlab
* github

### CODE REVIEW

File: app/DoctrineMigrations/Version20160430144756.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE project ADD namespace VARCHAR(255) NOT NULL');
```
Comment (tolry) [inline app/DoctrineMigrations/Version20160430144756.php]: could you add the appropriate update statement here to convert existing data?

File: src/AppBundle/Event/Subscriber/BadgeSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        $this->badgeGenerator = $badgeGenerator;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::NEW_MERGE_REQUEST => array('onNewMergeRequest', 0)
```
Comment (larsborn) [inline src/AppBundle/Event/Subscriber/BadgeSubscriber.php]: one line?

File: src/AppBundle/Events.php [hunk 3]
```diff
@@ -29,4 +29,9 @@
      * @var string
      */
     const WEBHOOK = 'simpspector.webhook';
+
+    /**
+     * 
+     */
+    const NEW_MERGE_REQUEST = 'simpspector.new_merge_request';
 }
```
Comment (larsborn) [inline src/AppBundle/Events.php]: missing newline

File: src/AppBundle/Provider/GithubProvider.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        /** @var Issue $mergeRequestApi */
+        $mergeRequestApi = $this->client->api('issue');
+        
+        $mergeRequestApi->comments()->create(
+            $project->getNamespace(),
+            $project->getName(),
+            $mergeRequest->getRemoteId(),
+            [
+                'body' => $comment
```
Comment (larsborn) [inline src/AppBundle/Provider/GithubProvider.php]: one line?

File: src/AppBundle/Provider/GitlabProvider.php [hunk 5]
```diff
@@ -27,40 +26,17 @@ public function __construct($url, $token)
...
-        $mergeRequest = new MergeRequest();
 
-        return $mergeRequest;
+        $mergeRequestApi->addComment(
+            $mergeRequest->getProject()->getRemoteId(),
+            $mergeRequest->getRemoteId(),
+            $comment
+        );
     }
 }
```
Comment (larsborn) [inline src/AppBundle/Provider/GitlabProvider.php]: missing newline

File: src/AppBundle/Provider/GithubProvider.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+     * @param MergeRequest $mergeRequest
+     * @param string $comment
+     * @throws \Github\Exception\MissingArgumentException
+     */
+    public function addMergeRequestComment(MergeRequest $mergeRequest, $comment)
+    {
+        $project = $mergeRequest->getProject();
+
+        /** @var Issue $mergeRequestApi */
+        $mergeRequestApi = $this->client->api('issue');
```
Comment (tolry) [inline src/AppBundle/Provider/GithubProvider.php]: maybe add a comment, why 'issue' is used here and not pr?

**Comment by tolry** [comment]
* usage of namespace in twig missing
* migration would be nice
* otherwise no objections from me

**Pull request merged**